resources/migration-guide-v2-v5.md

```diff
@@ -0,0 +1,402 @@
+- AsymmetricKey was splitted into to classes [PrivateKey](../src/types/keypair/PrivateKey.ts) and [PublicKey](../src/types/keypair/PublicKey.ts).
+
+#### Key Differences
+
+1. **Structure**:
+   - Old Keys class included functionality for both public and private keys in a single class.
+   - New structure separates concerns into distinct PublicKey and PrivateKey classes.
+2. **Enhanced Features**:
+   - The new classes provide extended functionality and better alignment with modern cryptographic standards.
+   - Includes support for additional key generation, validation, and serialization methods.
+
+#### Migration Example
+
+#### Old Syntax (2.x):
+
+```tsx
+import { Keys } from 'casper-js-sdk';
+
+const keyPair = Keys.Ed25519.new();
+const publicKey = keyPair.publicKey.toHex();
+const privateKey = keyPair.privateKey;
+```
+
+#### New Syntax (5.x):
+
+```tsx
+import { PublicKey, PrivateKey, KeyAlgorithm } from 'casper-js-sdk';
+
+// Generating a new key pair
+const keys = await PrivateKey.generate(KeyAlgorithm.ED25519);
+const publicKey = keys.publicKey.toHex();
+
+// Serialization
+console.log('Public Key (Hex):', publicKey.toHex());
+console.log('Private Key (PEM):', privateKey.toPem());
+```
+
+Refer to the [keypair folder](https://github.com/casper-ecosystem/casper-js-sdk/tree/feat-5.0.0/src/types/keypair) for detailed implementation and examples.
+
+---
+
+### 6. **`PurseIdentifier` Enum Replaced with a Class**
+
+### Changes
+
+- The PurseIdentifier enum has been replaced by a class with the same name, offering more flexibility and functionality.
+- The new class is located in the [request.ts](https://github.com/casper-ecosystem/casper-js-sdk/blob/feat-5.0.0/src/rpc/request.ts#L542) file.
+
+#### Migration Example
+
+#### Old Syntax (2.x):
+
+```tsx
+import { PurseIdentifier } from 'casper-js-sdk';
+
+const balanceByPublicKey = await client.queryBalance(
+  PurseIdentifier.MainPurseUnderPublicKey,
+  privateKey.publicKey.toHex(false)
+);
+
+const balanceByAccountHash = await client.queryBalance(
+  PurseIdentifier.MainPurseUnderAccountHash,
+  privateKey.publicKey.toAccountHashStr()
+);
+```
+
+#### New Syntax (5.x):
+
+```tsx
+import {
+  PurseIdentifier,
+  RpcClient,
+  HttpHandler
+} from 'casper-js-sdk';
+
+const rpcHandler = new HttpHandler('http://<Node Address>:7777/rpc');
+const rpcClient = new RpcClient(rpcHandler);
+
+try {
+  const balanceByPublicKey = await rpcClient.queryLatestBalance(
+    PurseIdentifier.fromPublicKey(privateKey.publicKey.toHex(false))
+  );
+
+  const balanceByAccountHash = await rpcClient.queryLatestBalance(
+    PurseIdentifier.fromAccountHash(
+      privateKey.publicKey.accountHash()
+    )
+  );
+} catch (e) {}
+```
+
+Explore the new PurseIdentifier class [here](https://github.com/casper-ecosystem/casper-js-sdk/blob/feat-5.0.0/src/rpc/request.ts#L542).
+
+---
+
+### 7. **`DeployUtil` to `Deploy`**
```

Comment:
Section 4 is `Changes to Deploy Management` and also tackles `DeployUtil` to `Deploy` change. Maybe we can merge the information here in the section above?